spring-mobile/src/main/java/org/togglz/spring/mobile/DeviceActivationStrategy.java

```diff
@@ -0,0 +1,67 @@
+package org.togglz.spring.mobile;
+
+import org.springframework.mobile.device.Device;
+import org.springframework.mobile.device.DeviceUtils;
+import org.togglz.core.activation.Parameter;
+import org.togglz.core.activation.ParameterBuilder;
+import org.togglz.core.repository.FeatureState;
+import org.togglz.core.spi.ActivationStrategy;
+import org.togglz.core.user.FeatureUser;
+import org.togglz.servlet.util.HttpServletRequestHolder;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Activation strategy that will use the Device type used by client to decide if a feature is active or not.
+ * Based on spring-mobile http://projects.spring.io/spring-mobile/
+ *
+ * Created by achhabra on 10/17/16.
+ * @author Anupriya Chhabra
+ *
+ */
+public class DeviceActivationStrategy implements ActivationStrategy {
+    public static final String DESKTOP = "Desktop";
+    public static final String MOBILE = "Mobile";
+    public static final String TABLET = "Tablet";
+    public static final String ON = "on";
+    public static final String YES = "YES";
+    public static final String ID = "devicerollout";
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public String getName() {
+        return "Device Rollout";
+    }
+
+    @Override
+    public Parameter[] getParameters() {
+        return new Parameter[]{
+                ParameterBuilder.create(ON+DESKTOP).label("Turn this test ON for desktop").matching("(YES|NO)")
+                        .description("Test will be off by default enter 'YES' in box above to enable"),
+                ParameterBuilder.create(ON+TABLET).label("Turn this test ON for Tablet").matching("(YES|NO)")
+                        .description(
+                        "Test will be off by default enter 'YES' in box above to enable"),
+                ParameterBuilder.create(ON+MOBILE).label("Turn this test ON for Mobile").matching("(YES|NO)")
+                        .description(
+                        "Test will be off by default enter 'YES' in box above to enable")
+        };
+    }
+
+    @Override
+    public boolean isActive(FeatureState featureState, FeatureUser user) {
+
+        HttpServletRequest request = HttpServletRequestHolder.get();
+        if (request != null) {
+            Device device = DeviceUtils.getCurrentDevice(request);
```

Comment:
Security issue? No. getParameters must be public, but getParameterName should be private.